File: scripts/extract-temp.py
import os
import sys
import json
import pdfplumber
import pandas as pd

sys.stdout.reconfigure(encoding='utf-8')

all_data = {
    'movvi': [],
    'braspress': [],
    'mengue': []
}

DOWNLOADS_DIR = r"C:\Users\cr_bn\Downloads"

def find_file(keyword, ext):
    for f in os.listdir(DOWNLOADS_DIR):
        if keyword.lower() in f.lower() and f.lower().endswith(ext.lower()) and not f.startswith('~'):
            return os.path.join(DOWNLOADS_DIR, f)
    return None

def extract_movvi_xlsx():
    path = find_file("movvi", ".xlsx")
    if not path:
        print("Movvi XLSX not found")
        return
    print(f"\n--- PARSING {os.path.basename(path)} ---")
    df = pd.read_excel(path, sheet_name=0, header=None)
    
    for i, row in df.iterrows():
        all_data['movvi'].append([x if pd.notna(x) else None for x in row.tolist()])

def extract_pdf_tables(keyword):
    path = find_file(keyword, ".pdf")
    if not path:
        print(f"{keyword} PDF not found")
        return
    
    print(f"\n--- PARSING {os.path.basename(path)} ---")
    try:
        with pdfplumber.open(path) as pdf:
            for page in pdf.pages:
                tables = page.extract_tables()
                for table in tables:
                    for row in table:
                        if any(x is not None for x in row):
                            # clean up newlines in cells
                            clean_row = [str(x).replace('\n', ' ').strip() if x is not None else None for x in row]
                            all_data[keyword].append(clean_row)
    except Exception as e:
        print(f"Error parsing {keyword}: {e}")

if __name__ == "__main__":
    extract_movvi_xlsx()
    extract_pdf_tables("braspress")
    extract_pdf_tables("mengue")
    
    with open("raw_tables.json", "w", encoding="utf-8") as f:
        json.dump(all_data, f, indent=2, default=str)
    print("Saved to raw_tables.json")

File: scripts/extract-temp.ts
import { createRequire } from 'module';
const require = createRequire(import.meta.url);
const fs = require('fs');
const path = require('path');
const xlsx = require('xlsx');
const pdfParse = require('pdf-parse');
console.log('type of pdfParse:', typeof pdfParse, 'keys:', Object.keys(pdfParse || {}));

async function parseMovviXlsx() {
  const filePath = 'C:/Users/cr_bn/Downloads/TABELA DE PREC\u0327OS - movvi.xlsx'; // the trailing zero width space or combined characters might be tricky, let's just use readdir
  // Let's resolve the actual file name
  const dir = 'C:/Users/cr_bn/Downloads';
  const files = fs.readdirSync(dir);
  const movviXlsxFile = files.find((f: any) => f.toLowerCase().includes('movvi.xlsx') && !f.startsWith('~'));

  if (!movviXlsxFile) {
    console.log('Movvi XLSX not found');
    return;
  }

  console.log(`\n\n--- PARSING ${movviXlsxFile} ---`);
  const workbook = xlsx.readFile(path.join(dir, movviXlsxFile));

  for (const sheetName of workbook.SheetNames) {
    console.log(`\nSheet: ${sheetName}`);
    const sheet = workbook.Sheets[sheetName];
    const json = xlsx.utils.sheet_to_json(sheet, { header: 1 });
    // Print first 30 rows
    for (let i = 0; i < Math.min(30, json.length); i++) {
      console.log(JSON.stringify(json[i]));
    }
  }
}

async function parseBraspressPdf() {
  const dir = 'C:/Users/cr_bn/Downloads';
  const files = fs.readdirSync(dir);
  const pdfFile = files.find((f: any) => f.toLowerCase().includes('braspress') && f.endsWith('.pdf'));

  if (!pdfFile) {
    console.log('Braspress PDF not found');
    return;
  }

  console.log(`\n\n--- PARSING ${pdfFile} ---`);
  const dataBuffer = fs.readFileSync(path.join(dir, pdfFile));
  const data = await pdfParse(dataBuffer);
  console.log(data.text.substring(0, 2000)); // First 2000 chars
}

async function parseMenguePdf() {
  const dir = 'C:/Users/cr_bn/Downloads';
  const files = fs.readdirSync(dir);
  const pdfFile = files.find((f: any) => f.toLowerCase().includes('mengue') && f.endsWith('.pdf'));

  if (!pdfFile) {
    console.log('Mengue PDF not found');
    return;
  }

  console.log(`\n\n--- PARSING ${pdfFile} ---`);
  const dataBuffer = fs.readFileSync(path.join(dir, pdfFile));
  const data = await pdfParse(dataBuffer);
  console.log(data.text.substring(0, 2000)); // First 2000 chars
}

async function parseMovviPdf() {
  const dir = 'C:/Users/cr_bn/Downloads';
  const files = fs.readdirSync(dir);
  const pdfFile = files.find((f: any) => f.toLowerCase().includes('movvi') && f.endsWith('.pdf'));

  if (!pdfFile) {
    console.log('Movvi PDF not found');
    return;
  }

  console.log(`\n\n--- PARSING ${pdfFile} ---`);
  const dataBuffer = fs.readFileSync(path.join(dir, pdfFile));
  const data = await pdfParse(dataBuffer);
  console.log(data.text.substring(0, 2000)); // First 2000 chars
}

async function main() {
  await parseMovviXlsx();
  await parseBraspressPdf();
  await parseMenguePdf();
  await parseMovviPdf();
}

main().catch(console.error);
